s3bench.go

```diff
@@ -104,19 +104,33 @@ func main() {
 	if !*skipCleanup {
 		fmt.Println()
 		fmt.Printf("Cleaning up %d objects...", *numSamples)
-		numDeleted := 0
 		svc := s3.New(session.New(), cfg)
-		for i := 0; i < *numSamples; i++ {
-			params := &s3.DeleteObjectInput{
-				Bucket: aws.String(*bucketName),
-				Key:    aws.String(fmt.Sprintf("%s%d", *objectNamePrefix, i)),
-			}
-			_, err := svc.DeleteObject(params)
-			if err == nil {
-				numDeleted++
+
+		numDeleted := 0
+		//max that can be deleted via DeleteObjects()
+		var chunkSize = 1000
+
+		keyList := make([]*s3.ObjectIdentifier, 0, chunkSize)
+		for i := 0; i < *numSamples; i++{
+			bar := s3.ObjectIdentifier{
+				Key: aws.String(fmt.Sprintf("%s%d", *objectNamePrefix, i)),
+				}
+			keyList = append(keyList, &bar)
+			if len(keyList) == chunkSize || i == *numSamples-1 {
+				fmt.Printf("Deleting %d objects\n", len(keyList))
```

Comment:
You could print the range and the status at the end of the line. If there is a failure it will identify which objects were not successfully cleaned up.

fmt.Printf("Deleting %d objects in {%d, %d}...", len(keyList), i-len(keyList), i)
if err != nil {
    fmt.Printf("Failed: %v\n", err)
} else {
   fmt.Printf("Succeeded\n")
}